### PR TITLE
GenAI: Add helper message to example prompt input that indicates whether requirements have been met or not.

### PR DESCRIPTION
**Pull request by cnbrenci**
This change adds an alert under the `Example Prompts and Topics` section of the `Publish` tab in Aichat levels. This alert appears when there are no prompts added and disappears when at least 1 prompt has been added.

https://github.com/user-attachments/assets/3655c1d7-f8ea-4b42-8bbc-36aef5c78824

It also tweaks the spacing around the "Added" items to match up better with the figma design. (follow-up from https://github.com/code-dot-org/code-dot-org/pull/60461#discussion_r1737027249)

Figma design:
image

Comparing prod to local:
- Removed extra space above `Add` button and below `Added` header
- Changed padding to be around the text inside the container rather than around the container, making the container full width and giving space between text and `x` button
- Changed `x` button to appear at the top right rather than centered on the right side.

https://github.com/user-attachments/assets/1055566c-2dc4-4229-906e-da288b81d6aa

## Links

- jira ticket: [Add helper message to example prompt input that indicates whether requirements have been met or not.](https://codedotorg.atlassian.net/browse/LABS-875)

## Testing story

- Tested locally the css updates with Retrieval and Publish tabs. 
- Tested the validation message only shows on the Publish tab and only when there are no added prompts, on first load and while interacting with the control.

### CODE REVIEW

File: apps/src/aichat/views/model-customization-workspace.module.scss [hunk 1]
```diff
@@ -185,17 +185,20 @@ $default-spacing: 4px;
...
-  padding: 4px 12px;
+  align-items: start;
   border-radius: 4px;
-  border-color: $light_gray_500;
   border: 1px solid;
   color: $light_gray_500;
+  border-color: $neutral_dark20;
+}
+
+.itemContainer span {
```
Comment (fisher-alice) [inline apps/src/aichat/views/model-customization-workspace.module.scss]: Nit: could add style rule within `.itemContainer` above
```
.itemContainer {
   ...
   > span {
      ...
   }
}

File: apps/src/aichat/views/modelCustomization/ExampleTopicsInputs.tsx [hunk 2]
```diff
@@ -39,6 +53,7 @@ const ExampleTopicsInputs: React.FunctionComponent<{
       isReadOnly={readOnly}
       hideInputBoxWhenReadOnly={false}
       onUpdateItems={onUpdateItems}
+      validationAlert={topics?.length <= 0 ? validationAlert : undefined}
```
Comment (fisher-alice) [inline apps/src/aichat/views/modelCustomization/ExampleTopicsInputs.tsx]: Nit: can length be `< 0` ?

File: apps/src/aichat/views/modelCustomization/ExampleTopicsInputs.tsx [hunk 3]
```diff
@@ -29,6 +32,17 @@ const ExampleTopicsInputs: React.FunctionComponent<{
     [dispatch]
   );
 
+  const validationAlert = useMemo(() => {
```
Comment (fisher-alice) [inline apps/src/aichat/views/modelCustomization/ExampleTopicsInputs.tsx:35]: This definitely works! But it reminds me of a comment I received from @thomasoniii at https://github.com/code-dot-org/code-dot-org/pull/60483#discussion_r1722172240.
I wonder if `ValidationAlert` could be defined as an internal component within `MultiInputCustomization` and perhaps the prop that is passed could be ~~is whether we want to display~~ the validation string displayed in the `ValidationAlert` (if not read-only).
I had to kinda think through/remember where else `MultiInputCustomization` is used (in `RetrievalCustomization`).

**Review by fisher-alice — APPROVED** [review]
Love the UI updates and how validation alert is removed when fulfilled 🎉 ! Just left a couple minor suggestions to consider. Thanks!

File: apps/src/aichat/views/modelCustomization/ExampleTopicsInputs.tsx (hunk 3 above)
Comment (cnbrenci) [reply apps/src/aichat/views/modelCustomization/ExampleTopicsInputs.tsx:35]: Chatted offline and going to do this refactor as a quick follow-up! It's a really good idea and will be a good little chunk of work I can do on my flight later :D

**Pull request merged**